base/strings/basic.jl

```diff
@@ -140,6 +140,32 @@ prevind(s::AbstractArray   , i::Integer) = i-1
 nextind(s::DirectIndexString, i::Integer) = i+1
 nextind(s::AbstractArray   , i::Integer) = i+1
 
+function prevind(s::String, i::Integer)
+    data = s.data
+    stop = endof(s.data)
+    if i > stop
+        return endof(s)
+    end
+    i -= oftype(i, 1)
```

Comment:
I don't think you should call `oftype` here. The index type for `String` is `Int` at the moment, we only accept `Integer` as input for convenience. Anyway, it doesn't make sense to be more general here than for `AsbtractString` below.